week3.ipynb

```diff
@@ -164,290 +155,122 @@
-    "print(periodic_table.columns.get_loc('NumberofValence'))\n",
-    "print(periodic_table.columns.get_loc('Element'))"
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# fill NaN with 0\n",
+    "periodic_table_no_nans = periodic_table.fillna(str(0))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Dataframes are great, but can be a little tricky to work with when using conditional statements, such as in `for` and `while` loops. For this exercise, we'll just convert our dataframe into a now-familiar numpy array."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 42,
-   "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "[]\n"
-     ]
-    }
-   ],
-   "source": [
-    "# let's use a for loop in combination with if-else statements to pull out a class of elements\n",
-    "# we know that if an element has a valence number of 1, then it must be an alkali metal. \n",
-    "# Hydrogen is an exception to this; we'll make sure to include that in our statements.\n",
-    "# from the dataframe above, we can see that the last column, \"NumberofValence,\" is what we're interested in\n",
-    "\n",
-    "alkali_metals = []\n",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# change dataframe to array\n",
+    "periodic_table_array = periodic_table_no_nans.to_numpy()\n",
     "\n",
-    "for i in range (len(periodic_table)):\n",
-    "    if (periodic_table_array[i, 27] == 1.0) and (periodic_table_array[i, 1] != 'Hydrogen'):\n",
-    "        np.append(alkali_metals, periodic_table_array[i, :])\n",
-    "    \n",
-    "print(alkali_metals)"
+    "# now if we inspect all of the columns in the first row (Row 0), we can see the name of each column\n",
+    "print(periodic_table_array[0, :])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The above print statement does return the information we wanted, but it's messy. Next, we'll use a `for` loop to neatly label the index of each column. "
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
-   "source": []
+   "source": [
+    "# for loop to clearly label the index of each column\n",
+    "\n",
+    "for i in range(len(periodic_table_array[0])):\n",
+    "    print(i, periodic_table_array[0, i])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Great! We can now easily index our array for any of the elemental properties presented here by using the index that appears next to the property."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we'll see how to use a `while` loop with `if-elif-else` statements to pull out chunks of data that meet a certain condition.\n",
```

Comment:
Is there an advantage to using a while loop here instead of a "for i in range(1, 118)"?

Perhaps add a statement comparing/contrasting these.

Overall I really enjoyed this exercise!